uber/models.py

```diff
@@ -946,14 +946,14 @@ def table_cost(self):
 
     @property
     def new_badge_cost(self):
-        return c.DEALER_BADGE_PRICE if self.is_dealer else c.get_group_price(sa.localized_now())
+        return c.DEALER_BADGE_PRICE if self.is_dealer else c.get_group_price()
 
     @cost_property
     def badge_cost(self):
         total = 0
         for attendee in self.attendees:
             if attendee.paid == c.PAID_BY_GROUP:
-                total += c.DEALER_BADGE_PRICE if attendee.is_dealer else c.get_group_price(attendee.registered)
+                total += c.DEALER_BADGE_PRICE if attendee.is_dealer else max(0, attendee.badge_cost - c.GROUP_DISCOUNT)
```

Comment:
This is the major change of this PR - instead of calculating their badge cost every time, groups will now look to their attendees' badge costs. This allows us to set `overridden_price` on each individual badge, preserving its cost in spite of price bumps based on bucket-based pricing. It should also allow us to do things like have minors in groups.